src/cluster.cc

```diff
@@ -53,6 +55,55 @@ Status Cluster::SetNodeId(std::string node_id) {
   return Status::OK();
 }
 
+// Set the slot to the node if new version is current version +1. It is useful
+// when we scale cluster avoid too many big messages, since we only update one
+// slot distribution and there are 16384 slot in our design.
+//
+// The reason why the new version MUST be +1 of current version is that,
+// the command changes topology based on specific topology (also means specific
+// version), we must guarantee current topology is exactly expected, otherwise,
+// this update may make topology corrupt, so base topology version is very important.
+// This is different with CLUSTERX SETNODES commands because it uses new version
+// topology to cover current version, it allows kvrocks nodes lost some topology
+// updates since of network failure, it is state instead of operation.
+Status Cluster::SetSlot(int slot, std::string node_id, int64_t new_version) {
+  // Parameters check
+  if (new_version <= 0 || new_version != version_ + 1) {
+    return Status(Status::NotOK, "Invalid cluster version");
+  }
+  if (!IsValidSlot(slot)) {
+    return Status(Status::NotOK, "Invalid slot id");
+  }
+  if (node_id.size() != kClusterNodeIdLen)  {
+    return Status(Status::NotOK, "Invalid node id");
+  }
+
+  // Get the node which we want to assign a slot into it
+  std::shared_ptr<ClusterNode> to_assign_node = nodes_[node_id];
+  if (to_assign_node == nullptr) {
+    return Status(Status::NotOK, "No this node in the cluster");
+  }
+  if (to_assign_node->role_ != kClusterMaster) {
+    return Status(Status::NotOK, "The node is not the master");
+  }
+
+  // Update version
+  version_ = new_version;
```

Comment:
We need to mark the clusterx command as exclusive to prevent data race?

Reply:
yes, for `clusterx` command, some subcommands already have `exclusive` attribute, `setslot` is one of them, please see `Cluster::SubCommandIsExecExclusive`

Reply:
ok, I forgot this point, only have a look at the command table.